butterknife-compiler/src/main/java/butterknife/compiler/BindingClass.java

```diff
@@ -259,6 +340,12 @@ private void addMethodBindings(MethodSpec.Builder result, ViewBindings bindings)
       result.beginControlFlow("if (view != null)");
     }
 
+    // Add the view reference to the unbinder.
+    if (unbinderName != null) {
+      result.addStatement("((UnbinderImpl) target.$L).$L = view", unbinderName,
```

Comment:
Also not a fan of casting this type every single time. I'd prefer to create it locally, assign all the views, and only set on the `target` at the end.